File: app/src/main/java/me/aluceps/sandbox/api/ConnpassClient.java
package me.aluceps.sandbox.api;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Single;
import me.aluceps.sandbox.api.service.ConnpassService;
import me.aluceps.sandbox.model.ConnpassEvent;

@Singleton
public class ConnpassClient {

    private final ConnpassService connpassService;

    @Inject
    public ConnpassClient(ConnpassService connpassService) {
        this.connpassService = connpassService;
    }

    public Single<ConnpassEvent> events() {
        return connpassService.events();
    }
}

File: app/src/main/java/me/aluceps/sandbox/api/ConnpassClient.kt
package me.aluceps.sandbox.api

import io.reactivex.Single
import me.aluceps.sandbox.api.service.ConnpassService
import me.aluceps.sandbox.model.ConnpassEvent
import me.aluceps.sandbox.model.RequestParams
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ConnpassClient @Inject
constructor(private val service: ConnpassService) {

    fun events(params: RequestParams): Single<ConnpassEvent> =
            service.events(params.offset, params.limit)
}
